src/puppetlabs/puppetdb/scf/storage.clj

```diff
@@ -1172,6 +1171,8 @@
   the specified date/time."
   [time]
   {:pre [(kitchensink/datetime? time)]}
+  (sql/update-values :certnames ["latest_report in (select id from reports where end_time < ?)" (to-timestamp time)]
+                     {:latest_report nil})
```

Comment:
Gotcha, is it possible to do the cascade for postgresql, and the nullification here for hsqldb only? Would mean if we retire hsqldb it would be more obvious for us (since grep would get this) + postgresql gets the better behaviour. Also a comment or update to the docstring as to the 'why' for this update would be useful.